test/integration/targets/iosxr_system/tests/netconf/set_hostname.yaml

```diff
@@ -8,11 +8,13 @@
       lines: hostname switch
       match: none
       provider: "{{ cli }}"
```

Comment:
provider is not required if connection is set to network_cli in task

Reply:
Same playbooks are used with `connection: local` and `connection:network_cli`. It will be ignored with connection: `network_cli`

Reply:
I thought `connection:` here would override the one in the tasks directory.

Reply:
@kedarX We can manage that at task level. Refer to https://github.com/ansible/ansible/blob/devel/test/integration/targets/nxos_config/tasks/cli.yaml#L30

Reply:
@trishnaguha if you have mix of cli and netconf tasks, include doesn't do the magic. It has highest precedence. So task level arg is required.

Reply:
@gundalow yes, `include` `ansible_connection` will over-ride task level connection. For netconf TCs, I am not using `include`